packages/app-mobile/components/ExtendedWebView/index.jest.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+
+import {
+	forwardRef, Ref, useEffect, useImperativeHandle, useMemo, useRef,
+} from 'react';
+
+import { View } from 'react-native';
+import Logger from '@joplin/utils/Logger';
+import { Props, WebViewControl } from './types';
+import { JSDOM } from 'jsdom';
+
+const logger = Logger.create('ExtendedWebView');
+
+const ExtendedWebView = (props: Props, ref: Ref<WebViewControl>) => {
+	const dom = useMemo(() => {
+		// Note: Adding `runScripts: 'dangerously'` to allow running inline <script></script>s.
+		// Use with caution.
+		return new JSDOM(props.html, { runScripts: 'dangerously' });
+	}, [props.html]);
+
+	useImperativeHandle(ref, (): WebViewControl => {
+		const result = {
+			injectJS(js: string) {
+				return dom.window.eval(js);
+			},
+			postMessage(message: unknown) {
+				const messageEventContent = {
+					data: message,
+					source: 'react-native',
+				};
+				return dom.window.eval(`
+					window.dispatchEvent(
+						new MessageEvent('message', ${JSON.stringify(messageEventContent)}),
+					);
+				`);
+			},
+
+			// Testing only
+			get document() {
+				return dom.window.document;
+			},
+		};
+		return result;
+	}, [dom]);
+
+	const onMessageRef = useRef(props.onMessage);
+	onMessageRef.current = props.onMessage;
+
+	// Don't re-load when injected JS changes. This should match the behavior of the native webview.
+	const injectedJavaScriptRef = useRef(props.injectedJavaScript);
+	injectedJavaScriptRef.current = props.injectedJavaScript;
+
+	useEffect(() => {
+		dom.window.eval(`
+			window.setWebViewApi = (api) => {
+				window.ReactNativeWebView = api;
+			};
+		`);
+		dom.window.setWebViewApi({
+			postMessage: (message: unknown) => {
+				logger.debug('Got message', message);
+				onMessageRef.current({ nativeEvent: { data: message } });
+			},
+		});
+
+		dom.window.eval(injectedJavaScriptRef.current);
+	}, [dom]);
+
+
+	const onLoadEndRef = useRef(props.onLoadEnd);
+	onLoadEndRef.current = props.onLoadEnd;
+	const onLoadStartRef = useRef(props.onLoadStart);
+	onLoadStartRef.current = props.onLoadStart;
+
+	useEffect(() => {
+		logger.debug(`DOM at ${dom.window?.location?.href} is reloading.`);
+		onLoadStartRef.current?.();
+		onLoadEndRef.current?.();
+	}, [dom]);
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- HACK: Allow wrapper testing logic to access the DOM.
+	const additionalProps: any = { document: dom?.window?.document };
+	return (
+		<View style={props.style} testID={props.testID} {...additionalProps}/>
+	);
```

Comment:
Currently, the virtual DOM for this mock is accessed with code similar to the following:
```ts
// Gets the <View></View> that is assigned a testID above:
const webviewContent = await screen.findByTestId('NoteBodyViewer');

// Get the `document` prop provided to the <View> through `additionalProps`:
const doc = webviewContent.props.document;

// Use the `document`
expect(doc.querySelector('h1').textContent).toBe('something'));
```

Above, `document` is retrieved through the `props` passed to the `<View>` object.

Previously, I was trying to access `document` through the `ref` passed to the parent of the `ExtendedWebView`. This, however, 1) [seems to involve tree navigation](https://callstack.github.io/react-native-testing-library/docs/advanced/testing-env#tree-navigation) and 2) `(await screen.findByTestId('id')).parent.props.ref` does not seem to contain the `ref` forwarded to the WebView's parent.